config-controller/internal/controller/policy_controller.go

```diff
@@ -103,44 +102,44 @@ func (r *SecurityPolicyReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 
 	// Non deletion case, so either an update or create of a policy
 	desiredState := policyCR.Spec.ToProtobuf()
-	desiredState.Name = policyCR.GetName()
-	desiredState.Source = storage.PolicySource_DECLARATIVE
+	desiredState.Id = policyCR.Status.PolicyId
 
-	existingPolicy, exists, err := r.PolicyClient.GetPolicy(ctx, req.Name)
+	_, exists, err := r.PolicyClient.GetPolicy(ctx, desiredState.GetName())
 	if err != nil {
 		return ctrl.Result{}, errors.Wrap(err, "Failed to fetch policy")
 	}
 
 	var retErr error
-	if exists {
-		desiredState.Id = existingPolicy.Id
-		if err = r.PolicyClient.UpdatePolicy(ctx, desiredState); err != nil {
-			desiredState.Id = existingPolicy.Id
+	if desiredState.Id != "" || exists {
```

Comment:
Documenting the possible states:

* `(desiredState.Id != "") == true` and `exists == true`: This CR has already been reconciled and is not being renamed.
* `(desiredState.Id != "") == false` and `exists == true`: This CR has never been successfully reconciled and a matching policy has been found in Central (`PUT`).
* `(desiredState.Id != "") == true` and `exists == false`: This CR has been reconciled before and is now being renamed.
* `(desiredState.Id != "") == false` and `exists == false`: This CR has never been successfully reconciled and a matching policy was NOT found in Central (`POST`).

(Note the behavior on bullet two is pending discussion later today)

Reply:
I'm not sure how exactly, but making these states more apparent in the code would be helpful.

Reply:
Reworked because cases have reduced